src/components/search/omnibox-directions.component.tsx

```diff
@@ -0,0 +1,288 @@
+import DateTimePicker from "@react-native-community/datetimepicker";
+import { POI, Location, MarkerLocation } from "../../types/main";
+import Autocomplete from "./autocomplete.component";
+import StartTravel from "./start-travel.component";
+import {
+  CONCORDIA_RED,
+  INPUT_BORDER_COLOR,
+  screenWidth
+} from "../../constants/style";
+import { getOmniboxAutoCompleteHeight } from "../../services/autoCompleteHeight.service";
+
+const getTimeToString = (date: Date) => {
+  const hours = `0${date.getHours().toString()}`;
+  const minutes = `0${date.getMinutes().toString()}`;
+
+  const hoursToShow = hours.substring(hours.length - 2, hours.length);
+  const minutessToShow = minutes.substring(minutes.length - 2, minutes.length);
+
+  return `${hoursToShow}:${minutessToShow}`;
+};
+/**
+ * the name and types of the properties types accepted
+ * by the OmniboxDirectionsProps component
+ */
+export interface OmniboxDirectionsProps {
+  destination: POI;
+  currentLocation: Location;
+  initialLocation: MarkerLocation;
+  setDestination: (poi: POI) => void;
+  setInitialLocation: (marker: MarkerLocation | POI | null) => void;
+  queryText: (
+    userInput: string,
+    setAutocomplete: (poi: POI[]) => void,
+    onChangeText: (string: string) => void
+  ) => void;
+  setIsDestinationFocused: (bool: boolean) => void;
+  isDestinationFocused: Boolean;
+  setStartTravelPlan: (bool: boolean) => void;
+}
+
+/**
+ * Displays the start and location with a departure time picker.
+ * Displays the transportation options
+ * @param destination
+ * @param initialLocation
+ * @param setDestination Function called to update destination
+ * @param setInitialLocation Function called to update initial Location
+ * @param queryText Function to query POI based on user input
+ */
+const OmniboxDirections = ({
+  destination,
+  currentLocation,
+  initialLocation,
+  setDestination,
+  setInitialLocation,
+  queryText,
+  setIsDestinationFocused,
+  isDestinationFocused,
+  setStartTravelPlan
+}: OmniboxDirectionsProps) => {
+  const [value, onChangeText] = React.useState(null);
+  const [destinationValue, setDestinationValue] = React.useState(
+    destination.displayName
+  );
+  const [autoCompleteValues, setAutocomplete] = React.useState(null);
+  const [autoCompleteValuesDest, setAutocompleteDest] = React.useState(null);
+  const [date, setDate] = React.useState(null);
+  const [showTimePicker, setshowTimePicker] = React.useState(false);
+
+  useEffect(() => {
+    if (initialLocation) onChangeText(initialLocation.displayName);
+    setAutocomplete(null);
+  }, [initialLocation]);
+
+  useEffect(() => {
+    setDestinationValue(destination.displayName);
+    setAutocompleteDest(null);
+  }, [destination]);
+
+  useEffect(() => {
+    if (currentLocation && !initialLocation) {
+      setInitialLocation({
+        id: "User Location",
+        displayName: "Current Location",
+        location: currentLocation
+      });
+    }
+  });
+
```

Comment:
![image](https://user-images.githubusercontent.com/33991121/77478340-32cd8900-6df4-11ea-99ac-69056729801b.png) I suspect this is caused by the ordering of react hooks, but the fixes I tried did not work. More on this here: https://reactjs.org/docs/hooks-rules.html Not sure if this will cause problems in the future.

Reply:
This was the only way I found to test useState hook and the fixes we've tried have not worked either. This increases our test coverage and should problems arise from this test in the future we can remove it.